AndelaOSP/Andela-Socials
# #158187265 Dockerize the application

#### What Does This PR Do?
- Setup docker for the application

#### Description Of Task To Be Completed
- Separate client and server in development application
- Create `docker-compose.yml` for development and production application
- Create `Dockerfile` for database service; for server and client services in development; and for application and nginx services in production
- Add setup scripts to the `Dockerfile`
- Add .dockerignore file for docker services
- Perform extensive testing on dockerized application

#### Any Background Context You Want To Provide?
- To have a unified environment for the application

#### How can this be manually tested?
- Ensure docker is installed locally on machine
- Ensure the environment variables in .env.sample are added in .env file in the root directory
- To build and spin up the docker containers, run `make build`
- To spin up the containers at a different time after building, run `make start`
- To stop all running containers, use the command `make stop`

#### What are the relevant github issues?
N/A

#### Screenshot(s)
![dockerize_app](https://user-images.githubusercontent.com/28448041/41293542-7f64a5d0-6e4d-11e8-99e3-c65987868f70.png)
